tests/integ/test_horovod_mx.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2017-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import json
+import os
+import tarfile
+from six.moves.urllib.parse import urlparse
+
+import boto3
+import pytest
+
+import sagemaker.utils
+import tests.integ as integ
+from sagemaker.mxnet import MXNet
+from tests.integ import timeout
+
+horovod_dir = os.path.join(os.path.dirname(__file__), "..", "data", "horovod")
+
+
+@pytest.mark.canary_quick
+def test_hvd_cpu(sagemaker_session, cpu_instance_type, tmpdir):
+    _create_and_fit_estimator(sagemaker_session, cpu_instance_type, tmpdir)
+
+
+@pytest.mark.canary_quick
+@pytest.mark.skipif(
+    integ.test_region() in integ.TRAINING_NO_P2_REGIONS, reason="no ml.p2 instances in this region"
+)
+def test_hvd_gpu(sagemaker_session, gpu_instance_type, tmpdir):
+    _create_and_fit_estimator(sagemaker_session, gpu_instance_type, tmpdir)
+
+
+@pytest.mark.local_mode
+@pytest.mark.parametrize("instances, processes", [[1, 2], (2, 1), (2, 2)])
+def test_horovod_local_mode(sagemaker_local_session, instances, processes, tmpdir):
```

Comment:
other than reading the output file, it doesn't seem like this test is providing much value for testing Python SDK functionality. would it be possible to instead put the assert at the end of the CPU test above?

Reply:
I'm following the same approach as horovod test for tensorflow in tests/integ/test_horovod.py
So for consistency purposes I'm sticking to a similar approach.

Reply:
I could see both the TF and this one being removed. testing for the processes being set up correctly, etc. should live in the framework toolkit repositories, and here we just need to make sure that the hyperparameters get passed along correctly.